pkg/resources/podAntiAffinity.go

```diff
@@ -0,0 +1,143 @@
+	isRequired, err := IsAntiAffinityRequired(ctx, client)
+	if err != nil {
+		isRequired = false
+	}
+
+	return func(obj metav1.Object, podTemplate *corev1.PodTemplateSpec) error {
+		labels := obj.GetLabels()
+		labelValue, ok := labels[labelMatch]
+		if !ok {
+			return fmt.Errorf("label %s not found in object", labelMatch)
+		}
+
+		podTemplate.Spec.Affinity = SelectAntiAffinityForCluster(isRequired, map[string]string{
+			labelMatch: labelValue,
+		})
+
+		return nil
+	}
+}
+
+// MultiAZAntiAffinityPreferred returns the affinity configuration to set the
+// preferred anti affinity by AZ on the given matchLabels
+func MultiAZAntiAffinityPreferred(matchLabels map[string]string) *corev1.Affinity {
+	return &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+				{
+					PodAffinityTerm: corev1.PodAffinityTerm{
+						LabelSelector: &v1.LabelSelector{
+							MatchLabels: matchLabels,
+						},
+						TopologyKey: ZoneLabel,
+					},
+					Weight: 100,
+				},
+			},
+		},
+	}
+}
+
+// MultiAZAntiAffinityRequired returns the affinity configuration to set the
+// required anti affinity by AZ on the given matchLabels
+func MultiAZAntiAffinityRequired(matchLabels map[string]string) *corev1.Affinity {
+	return &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &v1.LabelSelector{
+						MatchLabels: matchLabels,
+					},
+					TopologyKey: ZoneLabel,
+				},
+			},
+		},
+	}
+}
+
+// SelectAntiAffinityForCluster returns the affinity configuration for a cluster
+// given whether the rule is required or preferred
+func SelectAntiAffinityForCluster(required bool, matchLabels map[string]string) *corev1.Affinity {
+	if required {
+		return MultiAZAntiAffinityRequired(matchLabels)
+	}
+
+	return MultiAZAntiAffinityPreferred(matchLabels)
+}
+
+// IsAntiAffinityRequired checks whether the anti affinity rule must be set
+// to required or preferred.
+//
+// It currently checks the value of the FORCE_ZONE_DISTRIBUTION bool env var
+func IsAntiAffinityRequired(_ context.Context, _ k8sclient.Client) (bool, error) {
+	envValue, ok := os.LookupEnv(AntiAffinityRequiredEnvVar)
+	if !ok {
+		return false, nil
+	}
+
+	return strconv.ParseBool(envValue)
+}
+
+// IsMultiAZCluster checks if the cluster runs in multiple AZs, by retrieving
+// the nodes and checking that the `topology.kubernetes.io/zone` label is
+// the same across all
+func IsMultiAZCluster(ctx context.Context, client k8sclient.Client) (bool, error) {
+	// Get the list of nodes
+	nodeList := &corev1.NodeList{}
+	if err := client.List(ctx, nodeList); err != nil {
+		return false, err
+	}
+
+	// If there's no nodes, fail
+	if len(nodeList.Items) == 0 {
+		return false, fmt.Errorf("no nodes found")
+	}
+
+	// If there's only one node, directly return true
```

Comment:
Should this be `directly return false`?

Reply:
Oops yes, good catch. I'll fix it up with another typo that I found